.github/workflows/unit-tests.yml

```diff
@@ -0,0 +1,41 @@
+name: unit-tests
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+jobs:
+  unit-tests:
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: linux
+            runs-on: ubuntu-22.04
+            arch: x86_64
+
+    name: "${{ matrix.os }}-${{ matrix.arch }}"
+    runs-on: ${{ matrix.runs-on }}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+
+      - name: Install Nim
+        uses: iffy/install-nim@0ed218de08059d18507144e6aee70e6bad0bc148 # v5.0.3
+        with:
+          version: "binary:2.0.0"
+        env:
+          GITHUB_TOKEN: ${{ github.token }} # Avoid rate limiting.
+
+      - name: Install Nimble dependencies
+        run: nimble --accept install --depsOnly
+
+      - name: Run unit tests
+        run: nim r tests-nim/all_nim_tests.nim
```

Comment:
so far we normalize everything on docker image which has all deps pre-installed and is fetched from the github container registry as cache. lets use a consistent way to run nim things in all of CI

also we use `make` for triggering how tests are executed so lets add `make unit-tests` or similar target which will then use all the makefiles machinery to run tests as we currently do for rest of the tests

Reply:
as it adds it here, lets keep it consistent with existing patterns in the repo. currently:

* we use makefile for triggering tests in CI/etc as it can easily orchestrate prepping additional files in the repo.
* use docker for all dep orchestration. this allows to exactly replicate any CI setup locally with ease. also it makes it easy to manage/update all deps without changing system things. of course if people would like to run things on system they can but its more of an advanced mode and we cannot guarantee everything will work whereas our docker image we can. both @indecisivedragon and me exclusively use 🐳 for compiling chalk/running tests so lets follow the same here. makefile can easily call nimble `test` task if that simpler and you can use existing `chalk` compose service to leverage all pre-installed deps in there already - `docker compose run --rm chalk nimble test`.